Sources/MongoSwift/ReadPreference.swift

```diff
@@ -0,0 +1,164 @@
+import libmongoc
+
+/// A class to represent a MongoDB read preference.
+final public class ReadPreference {
+
+    /// An enumeration of possible ReadPreference modes.
+    public enum Mode: String {
+        /// See https://docs.mongodb.com/manual/reference/read-preference/
+        case primary
+        case primaryPreferred
+        case secondary
+        case secondaryPreferred
+        case nearest
+
+        internal var readMode: mongoc_read_mode_t {
+            switch self {
+            case .primary:
+                return MONGOC_READ_PRIMARY
+            case .primaryPreferred:
+                return MONGOC_READ_PRIMARY_PREFERRED
+            case .secondary:
+                return MONGOC_READ_SECONDARY
+            case .secondaryPreferred:
+                return MONGOC_READ_SECONDARY_PREFERRED
+            case .nearest:
+                return MONGOC_READ_NEAREST
+            }
+        }
+
+        internal init? (readMode: mongoc_read_mode_t) {
+            switch readMode {
+            case MONGOC_READ_PRIMARY:
+                self = .primary
+            case MONGOC_READ_PRIMARY_PREFERRED:
+                self = .primaryPreferred
+            case MONGOC_READ_SECONDARY:
+                self = .secondary
+            case MONGOC_READ_SECONDARY_PREFERRED:
+                self = .secondaryPreferred
+            case MONGOC_READ_NEAREST:
+                self = .nearest
+            default:
+                return nil
+            }
+        }
+    }
+
+    /// A pointer to a `mongoc_read_prefs_t`.
+    internal var _readPreference: OpaquePointer?
+
+    /// The mode of this `ReadPreference`
+    public var mode: Mode {
+        let readMode = mongoc_read_prefs_get_mode(self._readPreference)
+
+        guard let mode = Mode(readMode: readMode) else {
+            preconditionFailure("Unexpected read preference mode: \(readMode)")
+        }
+
+        return mode
+    }
+
+    /// The tags of this `ReadPreference`
+    public var tagSets: [Document] {
+        guard let bson = mongoc_read_prefs_get_tags(self._readPreference) else {
+            return []
+        }
+
+        let wrapped = Document(fromPointer: bson)
+
+        return wrapped.values as! [Document]
+    }
+
+    /// The maxStalenessSeconds of this `ReadPreference`
+    public var maxStalenessSeconds: Int64? {
+        let maxStalenessSeconds = mongoc_read_prefs_get_max_staleness_seconds(self._readPreference)
+
+        return maxStalenessSeconds == MONGOC_NO_MAX_STALENESS ? nil : maxStalenessSeconds
+    }
+
+    /**
+     * Initializes a `ReadPreference` from a `Mode`.
+     *
+     * - Parameters:
+     *   - mode: a `Mode`
+     *
+     * - Returns: a new `ReadPreference`
+     */
+    public init(_ mode: Mode) {
+        self._readPreference = mongoc_read_prefs_new(mode.readMode)
+    }
+
+    /**
```

Comment:
in this comment:
Int64 -> `Int64`
primary -> `.primary`
tagSets -> `tagSets`

Reply:
If `tagSets` should be monospaced as a parameter name, I assume `mode` and ``maxStalenessSeconds` should as well.

Reply:
ah yes, missed those. 

basically, if you look at how params show up in the docs ([here](https://mongodb.github.io/mongo-swift-driver/Classes/MongoDatabase.html#/s:10MongoSwift0A8DatabaseC10collectionAA0A10CollectionCyxGSS_xm8withTypeAA0E7OptionsVSg7optionstKs9DecodableRzs9EncodableRzlF) is one random example), it doesn't matter if there are backticks around the name in the parameter list because that get specially formatted, but anywhere else that a parameter, type etc is mentioned it's nice to have backticks so we get the monospace text.